vortex-array/src/array/constant/compute/mod.rs

```diff
@@ -48,6 +52,30 @@ impl ComputeVTable for ConstantEncoding {
     }
 }
 
+impl BinaryNumericFn<ConstantArray> for ConstantEncoding {
+    fn binary_numeric(
+        &self,
+        array: &ConstantArray,
+        rhs: &ArrayData,
+        op: NumericOperator,
+    ) -> VortexResult<Option<ArrayData>> {
+        let Some(rhs) = rhs.as_constant() else {
+            return Ok(None);
+        };
+
+        Ok(Some(
+            ConstantArray::new(
+                array
+                    .scalar()
+                    .as_primitive()
```

Comment:
might not be primitive?

Reply:
Should be for numeric compute. Implementations of compute functions are allowed to make assumptions that are validated by the entry point function (else everyone has to check index bounds and dtypes everywhere etc etc)